scene/gui/tree.cpp

```diff
@@ -4202,6 +4202,10 @@ void Tree::item_deselected(int p_column, TreeItem *p_item) {
 
 	if (select_mode == SELECT_MULTI || select_mode == SELECT_SINGLE) {
 		p_item->cells.write[p_column].selected = false;
+	} else {
+		for (int i = 0; i < p_item->cells.size(); i++) {
+			p_item->cells.write[i].selected = false;
+		}
```

Comment:
Maybe add `if (select_mode == SELECT_ROW)` after the `else` for better readability?